src/Index/src/Ingestor.cpp

```diff
@@ -178,6 +179,7 @@ namespace BitFunnel
     {
         ++m_documentCount;
         m_totalSourceByteSize += document.GetSourceByteSize();
+        m_maxDocId = id > m_maxDocId ? id : m_maxDocId;
```

Comment:
Is this really a safe multi-threaded operation? The other std::atomic variables use an std::atomic operation to do the read-modify-write. This max operation doesn't seem to be atomic.

Reply:
You're right, it is not safe. Use of compare_exchange_weak within a loop would make it safe.